src/server/pkg/collection/collection.go

```diff
@@ -58,7 +60,7 @@ type collection struct {
 }
 
 // NewCollection creates a new collection.
-func NewCollection(etcdClient *etcd.Client, prefix string, indexes []*Index, template proto.Message, keyCheck func(string) error, valCheck func(proto.Message) error) Collection {
+func NewCollection(etcdClient *etcd.Client, prefix string, indexes []*Index, template proto.Message, sortBy etcd.SortTarget, keyCheck func(string) error, valCheck func(proto.Message) error) Collection {
```

Comment:
It seems a little bit weird to have the sorting of collections as a property that's set in their construction, rather than something you pass in as an option to the methods for a couple of reasons:

- It's only applied to some methods and not all of them.
- It seems to be the same for every collection, and it would probably just lead to confusion if it was set to something else.

I think it'd probably be better to just make it a default that's hardcoded into the `Watch` methods rather than a configurable thing. That seems pretty reasonable to me because I think sorting by `ModRevision` is basically the only thing that makes sense in the `Watch` method.